linq/new-linq.ts

```diff
@@ -111,75 +137,62 @@ function isIterable<T>(source: any): source is Iterable<T> {
   return !!source && !!source[Symbol.iterator];
 }
 
-/** returns an IterableWithLinq<> for values in the collection 
- * 
- * @note - null/undefined/empty values are considered 'empty'
-*/
-export function values<K, T, TSrc extends (Array<T> | Dictionary<T> | Map<K, T>)>(...sources: Array<(Iterable<T> | Array<T> | Dictionary<T> | Map<K, T> | Set<T>) | null | undefined>): IterableWithLinq<T> {
-  return linqify({
-    [Symbol.iterator]: function* () {
-      for (const each of sources) {
-        yield* _values(each);
-      }
-    }
-  })
-}
-/** returns an IterableWithLinq<> for values in the collection */
-function _values<K, T, TSrc extends (Array<T> | Dictionary<T> | Map<K, T>)>(source: (Iterable<T> | Array<T> | Dictionary<T> | Map<K, T> | Set<T>) | null | undefined): IterableWithLinq<T> {
+export function values<K, T, TSrc extends (Array<T> | Dictionary<T> | Map<K, T>)>(source: (Iterable<T> | Array<T> | Dictionary<T> | Map<K, T> | Set<T>) | null | undefined): Iterable<T> {
   if (source) {
     // map
     if (source instanceof Map || source instanceof Set) {
-      return linqify(source.values());
+      return source.values();
     }
 
     // any iterable source
     if (isIterable(source)) {
-      return linqify(source);
+      return source;
     }
 
     // dictionary (object keys)
-    return linqify(function* () {
-      for (const key of keys(source)) {
-        const value = source[key];
-        if (typeof value !== 'function') {
-          yield value;
-        }
-      }
-    }());
+    return Object.values(source);
   }
 
   // null/undefined
-  return linqify([]);
+  return [];
+}
+export const linq ={
```

Comment:
Any particular reason for the namespacing?  Clarity/readability at call sites?

Reply:
To make using the `IterableWithLinq` functions a deliberate act -- the non namespaced versions are much more lightweigt.